exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -43,6 +43,11 @@ func (e *kafkaTracesProducer) tracesPusher(_ context.Context, td pdata.Traces) e
 	}
 	err = e.producer.SendMessages(messages)
 	if err != nil {
+		if value, ok := err.(sarama.ProducerErrors); ok {
+			for i := 0; i < len(value); i++ {
+				e.logger.Error("failed to export trace msg to kafka. ", zap.Error(value[i].Err))
+			}
+		}
```

Comment:
Few questions:
1. Kafka exporter uses queue-retry, do we want to print these errors even if there will be another retry?
2. I think we already print the error (but will be something like `fmt.Sprintf("kafka: Failed to deliver %d messages.", len(pe))` if that is the problem, maybe we just return a new "error" from here that instead of having that message actually concatenates all the errors? Then the queue-retry part will log the error for you.

These are all questions/ideas before we make a final call on what we want :) 

Thanks for the PR

Reply:
for question 1:when we use the kafka exporter in our prod  env, we have met a problem that if the metric msg's size is bigger than the max size we config in broker, it will be rejected by broker. The msg will be retried over and over again and will  never be succeeded to send to broker. Without the log we will never realize that a msg is discarded due  to its size. We wonder if there is something wrong with the msg itself. This problem had confused my team a couple of days.  btw, samara producer has a same config and will reject big size msg before send to broker, that is why I open another pr: #3580

Reply:
for question 2: I found that the queue_retry.go is in a common package `exporterhelper`, the error returned by sarama is the type `sarama.ProducerErrors`. I think maybe it is not good to covent the error's type to `sarama.ProducerErrors`  in package `exporterhelper`. But if u think that is ok, I will remove it to queue_retry.go.

Reply:
for question 2: maybe I should try to return a new error instead, you are right.